src/awkward/contents/unionarray.py

```diff
@@ -239,7 +239,16 @@ def simplified(
                                     contents[k].length,
                                 )
                             )
-                            contents[k] = contents[k].merge(inner_cont)
+                            old_parameters = contents[k]._parameters
+                            contents[k] = (
+                                contents[k]
+                                .merge(inner_cont)
+                                .copy(
+                                    parameters=ak._util.merge_parameters(
```

Comment:
`"__array__": "categorical"` can't be passed into `UnionArray.simplified` as a `parameters` argument. It will fail at some point (with a nice error message) because the output of `UnionArray.simplified` can't be an `IndexedArray` or `IndexedOptionArray`, and these are the only two node types that are allowed to have `"__array__": "categorical"` (now that wrong combinations are excluded in constructors).